connection.js

```diff
@@ -554,19 +554,29 @@ TChannelConnection.prototype.setupHandler = function setupHandler() {
     }
 };
 
+function getSocketBuffer(socket) {
```

Comment:
document the reason for this function and note which version each branch caters to.